core/services/pipeline/common.go

```diff
@@ -168,17 +170,30 @@ type JSONSerializable struct {
 	Null bool
 }
 
+// UnmarshalJSON implements custom unmarshaling logic
 func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 	if js == nil {
 		*js = JSONSerializable{}
 	}
+
 	return json.Unmarshal(bs, &js.Val)
 }
 
+// MarshalJSON implements custom marshaling logic
 func (js JSONSerializable) MarshalJSON() ([]byte, error) {
 	switch x := js.Val.(type) {
 	case []byte:
-		return json.Marshal(string(x))
+		// Don't need to HEX encode if it is a valid JSON string
+		if json.Valid(x) {
+			return json.Marshal(string(x))
+		}
+
+		// Don't need to HEX encode if it is already HEX encoded value
+		if _, err := strconv.ParseUint(string(x), 16, 64); err == nil {
```

Comment:
what happens if x > max uint64? it could still be a valid hex and this will error.

Reply:
Good question, updated the logic to do a regular HEX decode to check if the given string is already hex encoded